mlrun/datastore/azure_blob.py

```diff
@@ -29,47 +30,146 @@
+        Raises MLRunInvalidArgumentError if none of the connection details are available
+        based on do_connect in AzureBlobFileSystem:
+        https://github.com/fsspec/adlfs/blob/2023.9.0/adlfs/spec.py#L422
+        """
+        from azure.identity import ClientSecretCredential
+
+        storage_options = self.storage_options
+        connection_string = storage_options.get("connection_string")
+        client_name = storage_options.get("account_name")
+        account_key = storage_options.get("account_key")
+        sas_token = storage_options.get("sas_token")
+        client_id = storage_options.get("client_id")
+        credential = storage_options.get("credential")
+
+        credential_from_client_id = None
+        if (
+            credential is None
+            and account_key is None
+            and sas_token is None
+            and client_id is not None
+        ):
+            credential_from_client_id = ClientSecretCredential(
+                tenant_id=storage_options.get("tenant_id"),
+                client_id=client_id,
+                client_secret=storage_options.get("client_secret"),
+            )
+        try:
+            if connection_string is not None:
+                self._service_client = BlobServiceClient.from_connection_string(
+                    conn_str=connection_string,
+                    max_block_size=self.max_blocksize,
+                    max_single_put_size=self.max_single_put_size,
+                )
+            elif client_name is not None:
+                account_url: str = f"https://{client_name}.blob.core.windows.net"
+
+                creds = [
+                    credential_from_client_id or credential,
+                    account_key,
+                ]
+                if any(creds):
+                    first_cred = [item for item in creds if item is not None][0]
+                    self._service_client = BlobServiceClient(
+                        account_url=account_url,
+                        credential=first_cred,
+                        max_block_size=self.max_blocksize,
+                        max_single_put_size=self.max_single_put_size,
+                    )
+                elif sas_token is not None:
+                    if not sas_token.startswith("?"):
+                        sas_token = f"?{sas_token}"
+                    self._service_client = BlobServiceClient(
+                        account_url=account_url + sas_token,
+                        max_block_size=self.max_blocksize,
+                        max_single_put_size=self.max_single_put_size,
+                    )
+                else:
+                    # Fall back to anonymous login, and assume public container
+                    self._service_client = BlobServiceClient(
+                        account_url=account_url,
+                        max_block_size=self.max_blocksize,
+                        max_single_put_size=self.max_single_put_size,
+                    )
```

Comment:
I think it can be simplified quite a lot
```suggestion
                cred = credential_from_client_id or credential or account_key
                if not cred and sas_token is not None:
                    if not sas_token.startswith("?"):
                        sas_token = f"?{sas_token}"
                    account_url = account_url + sas_token
                self._service_client = BlobServiceClient(
                    account_url=account_url,
                    credential=cred,
                    max_block_size=self.max_blocksize,
                    max_single_put_size=self.max_single_put_size,
                )
```